pcdsdevices/lasers/qmini.py

```diff
@@ -83,6 +87,47 @@ class QminiSpectrometer(Device):
     fit_stdev = Cpt(EpicsSignalRO, ':STDEV', kind='config')
     fit_chisq = Cpt(EpicsSignalRO, ':CHISQ', kind='config')
 
+    # Save spectra functions
+    def save_data(self):
+        """
+        Save the wavelength and spectrum PVs to a text file
+        """
+        if len(self.file_dest.get().strip()) == 0:
+            # set a default destination for the file if you're lazy or give it whitespace
+            _file = os.getcwd() + '/' + self.name + time.strftime("_%Y-%m-%d_%H%M%S") + '.txt'
```

Comment:
Question: I noticed that you prepend a lot of function-scoped variables with a leading underscore. I'd previously only seen this as a convention for semi-private class members and function names that are intended not to be exported via import as per https://peps.python.org/pep-0008/.

Are there other reasons to name variables like this?

Reply:
I'm doing for (paranoid) privacy reasons on the calls. I'm probably being overly cautious, honestly, but it's a habit I've kept. I don't think there's going to be any conflicts (aside from the signal attribute name shenanigans that need it — as I found out the hard way)

Reply:
Can you expand a bit on privacy reasons? My question is mostly to learn

Reply:
I think, strictly speaking, `_var` is functionally a stylistic convention for saying "this `var` is private-ish, but I'm not gonna enforce it. Remember that you need to be janky to get here" vs. `__var` getting the built-in string mangling. 

Mostly useful for me in contexts where there are pretty commonly recycled var names (i.e. `result` , `data`, `output`, etc.) and I want to be explicit about which one I am messing with. That way I can have a parent function return a `result` dependent on an inner function call with output `_result` without getting too weird. Idk maybe I inherited some weird jank convention from `C` that I haven't unlearned.

Reply:
Gotcha, I do the same but only for data attributes e.g. `self._var`. I can see how it'd also be nice to indicate which variables are meant to persist through the function and which are e.g. loop-only. Thanks for humoring me.